### PR TITLE
Add Dns Resolve Host Count HealthCheck #684

### PR DESCRIPTION
**Pull request by CarlosLanderas**
Added new DNS healthchecks in Network package to allow configuring a target host and the minimum and maximum IP addresses that should be resolved for that host name.

```csharp
services
 .AddHealthChecks()
 .AddDnsResolveHostCountHealthCheck(setup =>
  {
     setup.AddHost("yourdomain.com", minHosts: 1, maxHosts: 3);
     setup.AddHost("otherdomain.com", minHosts: 1, maxHosts: 1);
  });
```

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**: #684

Please reference the issue this PR will close: #_[issue number]_

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

Please make sure you've completed the relevant tasks for this PR, out of the following list:

- [X] Code compiles correctly
- [X] Created/updated tests
- [X] Unit tests passing
- [X] End-to-end tests passing
- [ ] Extended the documentation
- [ ] Provided sample for the feature

### CODE REVIEW

**Comment by bakgerman** [comment]
yes that looks great!

**Comment by bakgerman** [comment]
So, if we don't care about max, only min, we could put MAX_INT as the value...or we could make max_hosts a nullable int, etc...

**Comment by CarlosLanderas** [comment]
@bakgerman changed.

Signature:

```csharp
  public DnsResolveCountOptions AddHost(string hostName, int minHosts, int? maxHosts)
        {
            if (!HostRegistrations.TryGetValue(hostName, out var _))
            {
                HostRegistrations.Add(hostName, (minHosts, maxHosts));
            }
            return this;
        }
```


Check:

```csharp
 if (totalAddresses < minHosts || (maxHosts.HasValue && totalAddresses > maxHosts.Value))
{
failedResolutions.Add((entry.Key, totalAddresses));
}
```

File: src/HealthChecks.Network/DnsResolveHostCountHealthCheck.cs [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+                foreach (var entry in _options.HostRegistrations)
+                {
+                    var (minHosts, maxHosts) = entry.Value;
+
+                    var ipAddresses = await Dns.GetHostAddressesAsync(entry.Key).WithCancellationTokenAsync(cancellationToken);
+                    var totalAddresses = ipAddresses.Count();
+
+
+
+                    if (totalAddresses < minHosts || (maxHosts.HasValue && totalAddresses > maxHosts.Value))
```
Comment (sungam3r) [inline src/HealthChecks.Network/DnsResolveHostCountHealthCheck.cs:36]:
```suggestion
                    if (totalAddresses < minHosts || totalAddresses > maxHosts.Value)
```

Comment (CarlosLanderas) [reply src/HealthChecks.Network/DnsResolveHostCountHealthCheck.cs:36]: mmm. maxHosts.Value will crash if it has no value?

Comment (sungam3r) [reply src/HealthChecks.Network/DnsResolveHostCountHealthCheck.cs:36]: Sorry, my typo, fixed.

**Pull request merged**